instaloader/structures.py

```diff
@@ -373,14 +375,20 @@ def is_video(self) -> bool:
     @property
     def video_url(self) -> Optional[str]:
         """URL of the video, or None."""
+        def get_size(url):
+            return requests.head(url).headers.get('Content-Length', 0), url
```

Comment:
Here we call requests directly and so bypass the HTTP headers and error handling that we usually have when doing a request. I think it would be better to rather have a method in `InstaloaderContext` to do that HEAD request with all the headers set, and ideally wrap HTTP error codes to exceptions. I think of something similar to `InstaloaderContext.get_raw(url: str) -> requests.Response`, with the difference that it does a HEAD request instead of a GET request, and that we use that new method instead of directly calling `requests.head()`. This would also allow to handle the case that a URL candidate yields a non-ok HTTP response.